CorsixTH/Lua/room.lua

```diff
@@ -573,16 +573,18 @@ function Room:onHumanoidLeave(humanoid)
     end
     -- Make patients leave the room if there are no longer enough staff
     if not self:testStaffCriteria(self:getRequiredStaffCriteria()) then
-      local patient_needs_to_reenter = false
+      local call_for_new_staff = false
       for room_humanoid in pairs(self.humanoids) do
-        if class.is(room_humanoid, Patient) and self:shouldHavePatientReenter(room_humanoid) then
-          self:makeHumanoidLeave(room_humanoid)
-          room_humanoid:queueAction(self:createEnterAction(room_humanoid))
-          patient_needs_to_reenter = true
+        if class.is(room_humanoid, Patient) and not room_humanoid:hasLeavingAction() then
+          call_for_new_staff = true
+          if self.room_info.id ~= "ward" then
+            self:makeHumanoidLeave(room_humanoid)
```

Comment:
Probably out of scope, but why does the room have a `makeHumanoidLeave' function, while the humanoid has a `hasLeavingAction` ?
Wouldn't it be more logical if the room asked the humanoid to leave? Ie actions are owned by the humanoid rather than owned by the room.

Reply:
A Humanoid:leaveRoom call would make sense, then have the patient determine if they need to use the screen to change back, instead of the makeHumanoidLeave and the makeHumanoidDressIfNecessaryThenLeave but rooms with screens normally have this driven by a callback or they are queued up upon getting sent to machine etc.

Reply:
Pass that into the humanoid function could work perhaps